waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/server/FederatedHMSHandler.java

```diff
@@ -1678,21 +1689,34 @@ public void update_master_key(int seq_number, String key) throws NoSuchObjectExc
   @Loggable(value = Loggable.DEBUG, skipResult = true, name = INVOCATION_LOG_NAME)
   public List<String> get_tables_by_type(String db_name, String pattern, String tableType)
       throws MetaException, TException {
-    DatabaseMapping mapping = databaseMappingService.databaseMapping(db_name);
+    String internalName = getDbInternalName(db_name);
+    DatabaseMapping mapping = databaseMappingService.databaseMapping(internalName);
     List<String> resultTables = mapping.getClient().get_tables_by_type(mapping.transformInboundDatabaseName(db_name), pattern, tableType);
-    List<String> result = databaseMappingService.filterTables(db_name, resultTables, mapping);
-    return mapping.getMetastoreFilter().filterTableNames(db_name, result);
+    List<String> result = databaseMappingService.filterTables(internalName, resultTables, mapping);
+    return mapping.getMetastoreFilter().filterTableNames(mapping.getCatalog(), internalName, result);
   }
 
   @Override
   @Loggable(value = Loggable.DEBUG, skipResult = true, name = INVOCATION_LOG_NAME)
   public GetTableResult get_table_req(GetTableRequest req) throws MetaException, NoSuchObjectException, TException {
-    DatabaseMapping mapping = getDbMappingAndCheckTableAllowed(req.getDbName(), req.getTblName());
+    String internal_name = getDbInternalName(req.getDbName());
+    DatabaseMapping mapping = getDbMappingAndCheckTableAllowed(internal_name, req.getTblName());
     GetTableResult result = mapping.getClient().get_table_req(mapping.transformInboundGetTableRequest(req));
     result.setTable(mapping.getMetastoreFilter().filterTable(result.getTable()));
     return mapping.transformOutboundGetTableResult(result);
   }
 
+  public static String getDbInternalName(String dbName)
+  {
+    try {
+      String internalName = parseDbName(dbName, null)[DB_NAME];
+      return internalName!=null ? internalName : "*";
```

Comment:
Not sure if defaulting to * is correct here, depends on what called this perhaps just return what was given, if null it's null the caller should deal with it.